pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -479,3 +480,38 @@ func (o *OpenIDCProvider) IsDisabledProvider() (bool, error) {
 	}
 	return !oidcConfig.Enabled, nil
 }
+
+func (o *OpenIDCProvider) getOIDCProvider(ctx context.Context, oidcConfig *v32.OIDCConfig) (*oidc.Provider, error) {
+	oidcFields := map[string]string{
+		client.OIDCConfigFieldIssuer:           oidcConfig.Issuer,
+		client.OIDCConfigFieldAuthEndpoint:     oidcConfig.AuthEndpoint,
+		client.OIDCConfigFieldTokenEndpoint:    oidcConfig.TokenEndpoint,
+		client.OIDCConfigFieldJWKSUrl:          oidcConfig.JWKSUrl,
+		client.OIDCConfigFieldUserInfoEndpoint: oidcConfig.UserInfoEndpoint,
+	}
+	var emptyFields []string
+	for key, value := range oidcFields {
+		if value == "" {
+			emptyFields = append(emptyFields, key)
+		}
+	}
+
+	// If all the fields are set, we will use them and manually specify each one.
+	// Otherwise, we will fall back to using just the issuer and the others will be determined by discovery.
+	if len(emptyFields) > 0 && slices.Contains(emptyFields, oidcFields[client.OIDCConfigFieldIssuer]) {
+		return nil, fmt.Errorf("unable to create OIDC provider.  The following fields are missing: %v", emptyFields)
```

Comment:
I don't think we need two spaces in here, and maybe we do a `strings.Join(emptyFields, ",")`?